astroid/objects.py

```diff
@@ -35,6 +37,11 @@
 
 objectmodel = util.lazy_import("interpreter.objectmodel")
 
+if sys.version_info >= (3, 8) or TYPE_CHECKING:
+    from functools import cached_property
+else:
+    from astroid.decorators import cachedproperty as cached_property
```

Comment:
Strange that `pylint: disable-next=ungrouped-imports` is needed in some cases but not all 🤔